metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -61,44 +60,59 @@ final class Embedded(
 }
 
 object Embedded {
-  def downloadSettings(
+  def fetchSettings(
       dependency: Dependency,
       scalaVersion: String
-  ): Settings =
-    new coursiersmall.Settings()
-      .withTtl(Some(Duration.Inf))
-      .withDependencies(List(dependency))
+  ): Fetch = {
+    val fetch = Fetch
+      .create()
+      .withDependencies(List(dependency): _*)
+      .addRepositories(
+        Repository.defaults().asScala.toList: _*
+      )
       .addRepositories(
         List(
-          coursiersmall.Repository.SonatypeReleases,
-          coursiersmall.Repository.SonatypeSnapshots
-        )
+          Repository.central(),
+          Repository.ivy2Local()
+        ): _*
       )
-      .withForceVersions(
+      .withDependencies(
         List(
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-library",
             scalaVersion
           ),
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-compiler",
             scalaVersion
           ),
-          new Dependency(
+          Dependency.of(
             "org.scala-lang",
             "scala-reflect",
             scalaVersion
           )
-        )
+        ): _*
       )
 
+    val envRepos = System
+      .getenv("COURSIER_REPOSITORIES")
+    if (envRepos.nonEmpty)
+      fetch.addRepositories(
+        envRepos
+          .split("""|""")
+          .map(MavenRepository.of)
+          .toList: _*
```

Comment:
can `toIndexedSeq` be used instead of `toList: _*` ? "Passing an explicit array value to a Scala varargs method is deprecated (since 2.13.0)"

Reply:
This is a Java vararg method. `toList:_*` is fine, we can deal with 2.13 deprecations another time